client/src/app/core/core-services/streaming-communication.service.ts

```diff
@@ -1,209 +1,158 @@
+            try {
+                await this.httpStreamService.connect(streamContainer, errorHandler);
+                success = true;
+            } catch (e) {
+                retries++;
+                if (streamContainer.stream) {
+                    streamContainer.stream.close();
+                    streamContainer.stream = null;
+                }
 
-    private hasError = false;
+                let goOffline = false;
+                if (retries < MAX_CONNECTION_RETRIES) {
+                    goOffline = !(await this.delayAndCheckReconnection());
+                } else {
+                    goOffline = true;
+                }
 
-    private _statuscode: number;
-    public get statuscode(): number {
-        return this._statuscode;
-    }
+                if (goOffline) {
+                    this.goOffline(streamContainer.endpoint);
+                    throw new OfflineError();
+                }
+            }
+        }
 
-    private _errorContent: any;
-    public get errorContent(): any {
-        return this._errorContent;
+        this.openStreams[streamContainer.id] = streamContainer;
+        return () => this.close(streamContainer);
     }
 
-    public readonly gotFirstResponse = new Deferred<boolean>();
-
-    /**
-     * This is the index where we checked, if there is a \n in the read buffer (event.partialText)
-     * This position is always >= contentStartIndex and is > contentStartIndex, if the message
-     * was too big to fit into one buffer. So we have just a partial message.
-     *
-     * The difference between this index and contentStartIndex is that this index remembers the position
-     * we checked for a \n which lay in the middle of the next JOSN-packet.
-     */
-    private checkedUntilIndex = 0;
+    private async handleError(streamContainer: StreamContainer, error: any): Promise<void> {
+        console.log('handle Error', streamContainer, streamContainer.stream, error);
+        streamContainer.stream.close();
+        streamContainer.stream = null;
+
+        streamContainer.hasErroredAmount++;
+        if (streamContainer.hasErroredAmount > MAX_STREAM_FAILURE_RETRIES) {
+            delete this.openStreams[streamContainer.id];
+            this.goOffline(streamContainer.endpoint);
+        } else {
+            // retry it after some time:
+            try {
+                if (await this.delayAndCheckReconnection()) {
+                    await this.connect(streamContainer);
+                }
+            } catch (e) {
+                // connectWithWrapper can throw an OfflineError. This one can be ignored.
+            }
+        }
+    }
 
     /**
-     * This index holds always the position of the current JOSN-packet, that we are receiving.
+     * Returns true, if a reconnect attempt should be done.
      */
-    private contentStartIndex = 0;
+    private async delayAndCheckReconnection(): Promise<boolean> {
+        const delay = Math.floor(Math.random() * 3000 + 2000);
```

Comment:
Is there a reason to randomize the delay instead of a constant?

Reply:
This was introduced to not instantly DDOS our servers on a mass reconnection. We tried to spread the requests a bit.